### PR TITLE
Gas station

### LINKED ISSUE #348: Gas station API
Create a new API that estimate the gas price so we avoid Metamask errors 

* Mainnet: https://safe-relay.gnosis.pm/api/v1/gas-station
* Rinkeby: https://safe-relay.staging.gnosisdev.com/api/v1/gas-station

For any other network, we can leave it undefined so it's estimated automatically by the wallet.

### PR DESCRIPTION
**Pull request by Velenir**
fetch **gasPrice** from gas station
only fetch when network or block has changed, cache otherwise

Maybe a better place for this logic would be in the upcoming Cache Layer. Then it could add `gasPrice` to `txOptionalParams` and pass onto the real `api.method`:

```ts
// in cache layer method
// only if method will send tx, not a call method

const txOptions = {
...txOptionalParams,
gasPrice: await fetchGasPrice(),
}

api.realMethod(args, txOptions)
```

Closes #348

### CODE REVIEW

File: src/api/wallet/WalletApi.ts [hunk 1]
```diff
@@ -279,7 +293,9 @@ export class WalletApiImpl implements WalletApi {
 
   /* ****************      Private Functions      **************** */
 
-  private async _notifyListeners(): Promise<void> {
+  private async _notifyListeners(blockchainUpdate?: BlockchainUpdatePrompt): Promise<void> {
```
Comment (Velenir) [inline src/api/wallet/WalletApi.ts]: same as update in #386

File: src/api/gasStation.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import { walletApi } from './'
```
Comment (anxolin) [inline src/api/gasStation.ts]: It looks problematic this dependency. If you depend in `WalletApi`, the gas station should be a "factory function" with `WalletApi` as one of it's params.

**Comment by Velenir** [comment]
Also not completely happy with almost-cyclic dependencies here. Injection in constructor does make , but wouldn't it be better as part of cache layer?

Will do injection for now.sense

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr409--dexreact.review.gnosisdev.com

File: src/api/gasStation.ts (hunk 2 above)
Comment (Velenir) [reply src/api/gasStation.ts]: Done

**Pull request merged**